### DISCUSSION TITLE
There is currently no equivalent to Copy-PnPFile in this library?

### DISCUSSION THREAD

**Discussion by horacioj** [pnp/pnpcore#715] [Q&A] [answered]
Just double-checking (I've read [#406](https://github.com/pnp/pnpcore/discussions/406)), that there is currently no way to copy a page between sites using this library? I.e. what PnP PowerShell's `Copy-PnPFile` does. 

Which would be the recommended way to accomplish this? 
Using raw REST? 

If this is the case, the authentication gathered via PnP.Core.Auth can be utilized somehow to perform raw REST requests, or a new authentication flow (for raw REST) should be implemented?

Thanks!

**Comment by jansenbe** [comment, marked as answer]
@horacioj : copying pages between sites is rather complex as you'd not only need to copy the page but also the relevant artefacts used by the page (or alternatively ensure the proper permissions are granted). Copying pages across sites is not something currently supported by the SharePoint UI either. If you're talking about copying files, then yes we still need to wrap the REST copy APIs in PnP Core (#406). If you can't wait for that work you can either make custom REST requests using PnP Core SDK (see https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html#making-a-custom-spo-rest-request) or alternatively update PnP Core SDK and create a PR to submit your work to the community.

**Reply by horacioj** [reply]
I guess using the (depreciated?)  CSOM library (which seems to have a way for [copying pages](https://sharepoint.stackexchange.com/a/205864)) isn't a good idea, right?

**Reply by jansenbe** [reply]
@horacioj : correct, that will not work for copying pages across sites.
